src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -0,0 +1,84 @@
+import { AfterViewChecked, Component, ElementRef, Input, OnChanges, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { ItemData } from '../../services/item-datasource.service';
+import { interval, merge, Observable } from 'rxjs';
+import { filter, map, startWith, switchMap } from 'rxjs/operators';
+import { SECONDS } from 'src/app/shared/helpers/duration';
+import { isNotUndefined } from 'src/app/shared/helpers/null-undefined-predicates';
+import { ItemTaskService } from '../../services/item-task.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { capitalize } from 'src/app/shared/helpers/case_conversion';
+import { ItemTaskInitService } from '../../services/item-task-init.service';
+import { ItemTaskAnswerService } from '../../services/item-task-answer.service';
+import { ItemTaskViewsService } from '../../services/item-task-views.service';
+
+const initialHeight = 1200;
+const heightSyncInterval = 0.2*SECONDS;
+
+interface TaskTab {
+  name: string,
+  view: string,
+}
+
+@Component({
+  selector: 'alg-item-display',
+  templateUrl: './item-display.component.html',
+  styleUrls: [ './item-display.component.scss' ],
+  providers: [ ItemTaskService, ItemTaskInitService, ItemTaskAnswerService, ItemTaskViewsService ],
+})
+export class ItemDisplayComponent implements OnInit, AfterViewChecked, OnChanges {
+  @Input() itemData?: ItemData;
+  @ViewChild('iframe') iframe?: ElementRef<HTMLIFrameElement>;
+
+  state$ = this.taskService.task$.pipe(mapToFetchState());
+
+  tabs$: Observable<TaskTab[]> = this.taskService.views$.pipe(
+    map(views => views.map(view => ({ view, name: this.getTabNameByView(view) }))),
+  );
+  activeTabView$ = this.taskService.activeView$;
+
+  iframeSrc$ = this.taskService.iframeSrc$;
+
+  // Start updating the iframe height to match the task's height
+  iframeHeight$ = merge(
+    this.taskService.task$.pipe(switchMap(task => interval(heightSyncInterval).pipe(switchMap(() => task.getHeight())))),
+    this.taskService.display$.pipe(map(({ height }) => height), filter(isNotUndefined)),
+  ).pipe(startWith(initialHeight), map(height => height + 40));
+
+  constructor(
+    private taskService: ItemTaskService,
+  ) {}
+
+  ngOnInit(): void {
+    if (!this.itemData) throw new Error('itemData must be set in ItemDisplayComponent');
+    this.taskService.configure(
+      this.itemData.route,
+      this.itemData.item.url ?? undefined,
+      this.itemData.route.attemptId ?? this.itemData.currentResult?.attemptId,
```

Comment:
Meaning the attempt may be `undefined`. 
But it should not be possible to load (or try to) a task if the user has no attempt yet or is not allowed to create attempt/result for this item. I haven't read the implementation of `configure` yet but in the objective of "failing fast", I don't think we should arrive up to here with a task not allowing attempt.

Actually, that's quite the same for url... I don't think it makes sense to load item display when there is no url (I might be wrong).

Reply:
Fixed